src/compiler/checker.ts

```diff
@@ -22935,7 +23023,10 @@ namespace ts {
                     (type === falseType || type === regularFalseType) ? TypeFacts.FalseStrictFacts : TypeFacts.TrueStrictFacts :
                     (type === falseType || type === regularFalseType) ? TypeFacts.FalseFacts : TypeFacts.TrueFacts;
             }
-            if (flags & TypeFlags.Object && !ignoreObjects) {
+            if (flags & TypeFlags.Object) {
+                if (ignoreObjects) {
+                    return TypeFacts.None;

```

Comment:
`ignoreObjects` is only going to be true if we are in the middle of computing type facts for an intersection type. In this case, the neutral value returned by `getTypeFacts` can't be `TypeFacts.All` (as it was before), because we are `or`ing the type facts. So it should be `TypeFacts.None`.